### DISCUSSION TITLE
How to model a relation where the same pair of objects can be related multiple times (with new properties each time)?

### DISCUSSION THREAD

**Discussion by HamishWHC** [geldata/gel#4923] [Q&A] [answered]
I've realised that I can't just add more links between the objects - it just updates the existing link. In this situation I am linking users to devices, each assignment begins and ends at a certain date and time, and hypothetically a device could be assigned to a user, reassigned to another, then assigned back to the same user. I'm assuming I just have to model the relation as an object itself (e.g. an Assignment type) and link devices to assignments and assignments to users, like you would model many-to-many in a traditional SQL DB, but I'm just wondering if anyone has an alternative that is more idiomatic to EdgeDB?

**Comment by msullivan** [comment, marked as answer]
Yeah, I think that's probably the best approach. 

You'll probably want to have the actual concrete links in the `Assignment` object (single links to both Device and User), because if you use a `multi` link from User to Assignment, you'll wind up having 4 tables in the underlying SQL database instead of 3.

You can then add some computed links to simplify the accesses

**Reply by HamishWHC** [reply]
Yeah that makes sense. Thanks for confirming.
